### DISCUSSION TITLE
Show more detailed errors to users?

### DISCUSSION THREAD

**Discussion by marty-sullivan** [danny-avila/LibreChat#4798] [Q&A]
Right now, when a user gets any sort of error from the custom endpoints we have set up, it just shows a generic "Here's the specific error we received: Please contact the admin..."

However, when I look in the logs, there will be a very detailed error message, typically about content filters being triggered. Is there a way to pass these error messages through to the user rather than asking them to contact the admin?

**Comment by danny-avila** [comment]
This is done by design, since error messages should not be shown to client: https://cwe.mitre.org/data/definitions/209.html

maybe this can be improved by allowing admin-set patterns, there is also a standard list of errors that are considered "safe" and are shown

**Reply by marty-sullivan** [reply]
I do think it would be good to have some kind of set of allowed error messages whitelist, or a way to distinguish between client-errors and server-errors.

We have given access to 2500 (and growing) users, so not being able to pass obvious client-side errors that the user can fix themselves is going to become a severe growing pain.
